lib/view/pages/stats/stats_genres.dart

```diff
@@ -81,10 +109,13 @@ class _GenresStatsPageState extends State<GenresStatsPage> {
                     ),
                     Padding(
                       padding: const EdgeInsetsDirectional.fromSTEB(0, 0, 20, 0),
-                      child: Icon(
-                        Icons.filter_alt,
-                        color: Theme.of(context).colorScheme.primary,
-                        size: 24,
+                      child: IconButton(
+                        icon: Icon(
+                          Icons.filter_alt,
+                          color: Theme.of(context).colorScheme.primary,
+                          size: 24,
+                        ),
+                        onPressed: _showTimeRangeDialog,
```

Comment:
Rather than using different functions to build different components, please create a component in the components directory and add the code there.

Reply:
I created a component for this